cmd/mysterium_server/command_run/factory.go

```diff
@@ -1,34 +1,74 @@
 package command_run
 
 import (
+	"github.com/ethereum/go-ethereum/accounts/keystore"
 	"github.com/mysterium/node/communication"
 	"github.com/mysterium/node/communication/nats_dialog"
 	"github.com/mysterium/node/communication/nats_discovery"
 	"github.com/mysterium/node/identity"
 	"github.com/mysterium/node/ipify"
 	"github.com/mysterium/node/nat"
 	"github.com/mysterium/node/openvpn"
+	openvpn_session "github.com/mysterium/node/openvpn/session"
 	"github.com/mysterium/node/server"
 	dto_discovery "github.com/mysterium/node/service_discovery/dto"
 	"github.com/mysterium/node/session"
-	"os"
 )
 
-func NewCommand(vpnMiddlewares ...openvpn.ManagementMiddleware) *CommandRun {
-	return &CommandRun{
-		Output:      os.Stdout,
-		OutputError: os.Stderr,
+func NewCommand(options CommandOptions) *CommandRun {
+	return NewCommandWith(
+		options,
+		server.NewClient(),
+		ipify.NewClient(),
+		nat.NewService(),
+	)
+}
+
+func NewCommandWith(
+	options CommandOptions,
+	mysteriumClient server.Client,
+	ipifyClient ipify.Client,
+	natService nat.NATService,
+) *CommandRun {
+
+	ks := keystore.NewKeyStore(options.DirectoryKeystore, keystore.StandardScryptN, keystore.StandardScryptP)
+	identityHandler := NewNodeIdentityHandler(
+		identity.NewIdentityManager(ks),
+		mysteriumClient,
+		options.DirectoryKeystore,
+	)
 
-		IpifyClient:     ipify.NewClient(),
-		MysteriumClient: server.NewClient(),
-		NatService:      nat.NewService(),
-		DialogWaiterFactory: func(identity identity.Identity) (communication.DialogWaiter, dto_discovery.Contact) {
+	return &CommandRun{
+		identitySelector: func() (identity.Identity, error) {
+			return identityHandler.Select(options.NodeKey)
+		},
+		ipifyClient:     ipifyClient,
+		mysteriumClient: mysteriumClient,
+		natService:      natService,
+		dialogWaiterFactory: func(identity identity.Identity) (communication.DialogWaiter, dto_discovery.Contact) {
 			address := nats_discovery.NewAddressForIdentity(identity)
 			return nats_dialog.NewDialogWaiter(address), address.GetContact()
 		},
-		SessionManager: &session.Manager{
-			Generator: &session.Generator{},
+		sessionManagerFactory: func(vpnServerIp string) session.ManagerInterface {
+			return openvpn_session.NewManager(openvpn.NewClientConfig(
+				vpnServerIp,
+				options.DirectoryConfig+"/ca.crt",
```

Comment:
Don't hardcode slashes - the code is not portable between OSes. https://golang.org/pkg/path/filepath/#Join is your friend here.

Reply:
Fixed in all codebase